### PR TITLE
Adds error screen when JSON RPC fails to connect

### LINKED ISSUE #49: Make sure that WebUI shows an error message when the RPC connection fails.
Currently when the RPC connection fails for whatever reason the WebUI will get stuck at `Loading...`
The only way to figure out that something failed is to open the console.

This should change and a proper message should be shown to the user, mentioning an issue when contacting the RPC endpoint.

### PR DESCRIPTION
**Pull request by kelsos**
image
image

Closes #49

### CODE REVIEW

File: src/app/services/shared.service.ts [hunk 1]
```diff
@@ -45,11 +54,31 @@ export class SharedService {
         this.toastrService.warning(message.description, message.title);
     }
 
+    public set status(value: ConnectivityStatus) {
```
Comment (andrevmatos) [inline src/app/services/shared.service.ts]: Why the getter/setter, if they're trivial and public? I think we don't need these javisms in TS, IMO ;)

Comment (andrevmatos) [reply src/app/services/shared.service.ts]: Besides, seems like it's only used in that `hasRpcError`. Wouldn't it be more simple just testing for the stack trace, as they're always set together?

File: src/app/services/shared.service.ts [hunk 2]
```diff
@@ -45,11 +54,31 @@ export class SharedService {
...
+
     requestStarted() {
         this.requestsSubject.next(+1);
     }
 
     requestFinished() {
         this.requestsSubject.next(-1);
     }
+
+    setStackTrace(stackTrace: Error | null) {
```
Comment (andrevmatos) [inline src/app/services/shared.service.ts]: Same setter/getter comment

File: src/app/services/shared.service.ts (hunk 1 above)
Comment (kelsos) [reply src/app/services/shared.service.ts]: I mostly had in mind to add some kind of connectivity report here afterwards. 
For example if the node stops but you still have the webui loaded in your browser.

However I can unify everything and then if there is the need I can modify.

File: src/app/services/shared.service.ts (hunk 2 above)
Comment (kelsos) [reply src/app/services/shared.service.ts]: I will still keep the `getStackTrace` mostly because it has some minor logic.

**Pull request merged**